### PR TITLE
165716787 Build Article Shared Components

### PR DESCRIPTION
**Pull request by micah-akpan**
#### Description
Adds Reusable Components (Article Cards)
#### Type of change
- [ ] Bug Fix (a non-breaking change which fixes an issue)
- [x] New Feature (a non-breaking change that adds functionality)
- [ ] New Chore (a non-breaking change that adds functionality)
- [ ] Breaking Change (a fix or feature which would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
#### How Has This Been Tested?
- [ ] End to End
- [ ] Integration
- [x] Unit test
#### Checklist:
- [x] Unit test
- [ ] Code coverage
#### Thought Process
1. Design Card Component that takes article as prop
2. Stub Card Component
3. Test Card Component that it renders correctly
4. Develop Card Component
5. Display article title, description, total read time on card
6. Test Card Component children
7. Refactor Components
#### Screenshot
- Article Primary Card
Screenshot 2019-05-13 at 11 28 46 AM
- Article Secondary Card
Screenshot 2019-05-13 at 11 29 00 AM

#### PT
[#165716787](https://www.pivotaltracker.com/story/show/165716787)

### CODE REVIEW

File: src/components/shared/Article/ArticlePrimaryCard.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          <h3>{article.title}</h3>
+          <p>{article.description}</p>
+          <span>{article.author}</span>
+          <span>
+            {parseArticleCreationDate(article.createdAt)}
+            &nbsp;&middot;
+          </span>&nbsp;
+          <ArticleReadTime data-testId={`article-read-time-${article.id}`}>
+            {article.totalReadTime} {article.totalReadTime > 1 ? 'mins' : 'min'} read
+          </ArticleReadTime>&nbsp;
```
Comment (chikeozulumba) [inline src/components/shared/Article/ArticlePrimaryCard.jsx]: @micah-akpan Why **&nbsp**;

File: src/components/shared/Article/ArticleSecondaryCard.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+`;
+
+const ArticleSecondaryCard = ({ article, count }) => (
+    <Grid.Row>
+      <StyledColumn width={4} verticalAlign="middle">
+        <StyledNumber>0{count + 1}</StyledNumber>
+      </StyledColumn>
+
+      <Grid.Column>
+        <h3>{article.title}</h3>
```
Comment (chikeozulumba) [inline src/components/shared/Article/ArticleSecondaryCard.jsx]: Could you have considered deep destructuring? @micah-akpan

File: src/utils/articles.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import moment from 'moment';
+
+export default {
```
Comment (chikeozulumba) [inline src/utils/articles.js:3]: @micah-akpan  could you elaborate on why you chose to use export default, and not export.
Also, why are you exporting an object

**Review by chikeozulumba — COMMENTED** [review]
Great work @micah-akpan 
Could you shed some clarity on my questions?

**Review by micah-akpan — COMMENTED** [review]
The functions inside this module are all related so I'm putting them here to export a single object as a namespace to all the functions. Do you think named exports would be better in this context?

File: src/components/shared/Article/ArticleSecondaryCard.jsx (hunk 2 above)
Comment (micah-akpan) [reply src/components/shared/Article/ArticleSecondaryCard.jsx]: Okay that's a good suggestion. I just didn't see the need at the time. A normal destructuring will work here. I will incorporate the suggested changes. Thanks

File: src/utils/articles.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    return largestReadCount;
+  },
+
+  /**
+   * @function getTrendingArticle
+   * @param {Array} allArticles
+   * @returns {Object} Returns an array with the highest read count value
+   */
+  getTrendingArticle(allArticles) {
+    return allArticles.find(article => article.readCount === this.getLargestReadCount(allArticles));
```
[the remark this answers is not in the document]
Comment (micah-akpan) [reply src/utils/articles.js:83]: This will be fixed in the next PR

File: src/utils/articles.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  },
+
+  /**
+   * @function getTopTrendingArticles
+   * @param {Array} articles
+   * @param {Number} maxReadCount
+   * @param {Number} number
+   * @returns {Array} Returns `number` most trending articles
+   */
+  getTopTrendingArticles(articles, maxReadCount = 100, number = 3) {
```
[the remark this answers is not in the document]
Comment (micah-akpan) [reply src/utils/articles.js:93]: This will be fixed in the next PR

File: src/utils/articles.js (hunk 3 above)
Comment (micah-akpan) [reply src/utils/articles.js:3]: The functions inside this module are all related so I'm putting them here to export a single object as a namespace to all the functions. Do you think named exports would be better in this context?

**Comment by Raymond-Osy** [comment]
I guess Hound CI has some configurations to be fixed, you might want to take note of it...

**Review by toluola — DISMISSED** [review]

**Review by Temmyogunbo — DISMISSED** [review]

**Review by Temmyogunbo — APPROVED** [review]

**Pull request merged**